### PR TITLE
Add table of contents

### PR DESCRIPTION
**Pull request by jog1t**
Closes GDT-87

### CODE REVIEW

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
Absolute legend.

A couple notes:

- The headings need to update as you scroll instead of the hash

- The media queries seem to not work correctly at a certain size



- I can't put my finger on the right solution, but the header on the right sidebar feels a bit off being misaligned with the text. We could either (a) align the header with the text, even though it looks weird or (b) remove the background on the buttons so we don't need padding around the buttons

- The header overlaps the main content oddly



- Needs to have h1, h2, and h3 with indent like [here](https://syntax.tailwindui.com/docs/understanding-caching)

Fyi we have a license to all of the tailwind templates: https://tailwindui.com/templates/syntax

**Comment by NathanFlurry** [comment]
@jog1t Awesome, you're probably right to roll this one out quickly. Let's fix #1 and #3 and we can merge this in asap.

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
@jog1t this is honestly _so_ cool. clicked around a few other docs pages, this is night and day for our docs. 🙌 

I have a bunch more issues:

**Scroll ToC**

If easy, can we make this automatically keep the current tab in the view when scrolling? Note how the table of contents active selector goes out of view here:

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/BIF3m0zL7mcrHUeMWh4D/bb669964-7ffb-41d4-a28c-77bf665d6dd6.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/BIF3m0zL7mcrHUeMWh4D/bb669964-7ffb-41d4-a28c-77bf665d6dd6.mov">Graphite Add Table of Contents.mov</video>

Going back to the GH actions, realizing that they actually do your original implementation (i.e. using the anchor), but imo this feels way more fluid.

**Tiny content area**

The content gets _super_ small on a small screen:



Compared to GH actions:



I'm wondering if we should collapse our left sidebar like they do or collapse the table of contents.

Related issues:

- https://linear.app/rivet-gg/issue/HUB-239/sidebar-nav-button-does-not-show-when-sidebar-is-hidden
- https://linear.app/rivet-gg/issue/HUB-240/header-logo-and-open-rivet-buttons-collapse-weirdly-on-smaller-screens

File: src/components/TableOfContents.jsx
Comment (AngelOnFira) [inline src/components/TableOfContents.jsx:1]: I found a bug where ToC will show sections that have been commented out. This is the code for the section:

```
{/* 
### Test locally

First, we can test that the game works locally.

Open the Rivet panel in the bottom left and click _Start Server_. Then click the
play button in the top right to start the client. You should be able to join the
lobby. */}
```

Comment (AngelOnFira) [inline src/components/TableOfContents.jsx:1]: Running into an issue on where the page will scroll to when clicking in the ToC

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
I ran into a few bugs to resolve 💯 sorry if either of them have already been brought up!

**Comment by NathanFlurry** [comment]
@jog1t cf pages failing

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
Getting this error:

**Review by NathanFlurry — APPROVED** [review]
dope

File: src/components/Code.jsx
Comment (AngelOnFira) [inline src/components/Code.jsx:1]: I'm still seeing some inconstancies on how it scrolls through on the right side, and when clicking on sections, it still doesn't seem to work all the time. I think it's good enough for now, and I can create separate tickets when I run into things.

**Review by AngelOnFira — APPROVED** [review]

**Pull request merged**